packages/plugin/webpack/src/WebpackConfig.ts

```diff
@@ -163,114 +180,122 @@ export default class WebpackConfigGenerator {
+        }
+      }
     }
+    const rendererConfigPromises = [
+      this.buildRendererConfig(entryPointsForTarget.web, RendererTarget.Web),
+      this.buildRendererConfig(entryPointsForTarget.electronRenderer, RendererTarget.ElectronRenderer),
+      this.buildRendererConfig(entryPointsForTarget.electronPreload, RendererTarget.ElectronPreload),
+      this.buildRendererConfig(entryPointsForTarget.sandboxedPreload, RendererTarget.SandboxedPreload),
+    ];
+
+    const rendererConfig = await Promise.all(rendererConfigPromises).then((res) => {
+      return res.filter(function isNotNull<T>(item: T | null): item is T {
+        return !!item;
+      });
+    });
+    return rendererConfig;
+  }
 
-    const rendererConfig = await this.resolveConfig(entryPoint.preload.config || this.pluginConfig.renderer.config);
-    const prefixedEntries = entryPoint.prefixedEntries || [];
-
-    return webpackMerge(
-      {
-        devtool: this.rendererSourceMapOption,
-        mode: this.mode,
-        entry: prefixedEntries.concat([entryPoint.preload.js]),
-        output: {
-          path: path.resolve(this.webpackDir, 'renderer', entryPoint.name),
-          filename: 'preload.js',
-        },
-        node: {
-          __dirname: false,
-          __filename: false,
-        },
+  buildRendererBaseConfig(target: RendererTarget): webpack.Configuration {
+    return {
+      target: rendererTargetToWebpackTarget(target),
+      devtool: this.rendererSourceMapOption,
+      mode: this.mode,
+      output: {
+        path: path.resolve(this.webpackDir, 'renderer'),
+        filename: '[name]/index.js',
+        globalObject: 'self',
+        ...(this.isProd ? {} : { publicPath: '/' }),
       },
-      rendererConfig || {},
-      { target: 'electron-preload' }
-    );
+      node: {
+        __dirname: false,
+        __filename: false,
+      },
+      plugins: [new AssetRelocatorPatch(this.isProd, !!this.pluginConfig.renderer.nodeIntegration)],
+    };
   }
 
-  async getRendererConfig(entryPoints: WebpackPluginEntryPoint[]): Promise<Configuration[]> {
+  async buildRendererConfig(entryPoints: WebpackPluginEntryPoint[], target: RendererTarget): Promise<Configuration | null> {
+    const entry: webpack.Entry = {};
+
+    if (entryPoints.length === 0) {
+      return null;
+    }
+    const baseConfig: webpack.Configuration = this.buildRendererBaseConfig(target);
     const rendererConfig = await this.resolveConfig(this.pluginConfig.renderer.config);
 
-    return entryPoints.map((entryPoint) => {
-      const baseConfig: webpack.Configuration = {
-        target: this.rendererTarget(entryPoint),
-        devtool: this.rendererSourceMapOption,
-        mode: this.mode,
+    if (target === RendererTarget.Web || target === RendererTarget.ElectronRenderer) {
+      for (const entryPoint of entryPoints) {
+        entry[entryPoint.name] = (entryPoint.prefixedEntries || []).concat([isPreloadOnly(entryPoint) ? entryPoint.preload.js : entryPoint.js]);
```

Comment:
Why is `preload` in here? Neither the `Web` nor `ElectronRenderer` targets should be considering preload values

Reply:
TS gets angry and complains that `entryPoint.js` doesn't exist on `WebpackPluginEntryPointPreloadOnly`, even though that should not have made it thrtough the if statement, but TS doesn't know that